services/updates-svc/stories/updates_test.go

```diff
@@ -0,0 +1,52 @@
+package stories
+
+import (
+	"common/auth"
+	"context"
+	pb "gen/services/updates_svc/v1"
+	"github.com/stretchr/testify/assert"
+	"io"
+	"testing"
+	"time"
+)
+
+func TestOpenAndClosingReceiveUpdatesStream(t *testing.T) {
+	ctx := context.Background()
+	updatesClient := updatesServiceClient()
+
+	req := &pb.ReceiveUpdatesRequest{}
+	stream, err := updatesClient.ReceiveUpdates(ctx, req)
+	assert.NoError(t, err)
+	assert.NoError(t, stream.CloseSend())
+}
+
+func TestAutoClosingWhenTokenExpiresReceiveUpdateStream(t *testing.T) {
+	ctx := context.Background()
+	updatesClient := updatesServiceClient()
+
+	auth.FakeTokenValidFor = time.Second * 3
+	upperBound := auth.FakeTokenValidFor + (time.Second * 1)
+	lowerBound := auth.FakeTokenValidFor - (time.Second * 1)
+
+	req := &pb.ReceiveUpdatesRequest{}
+
+	timeBeforeOpeningStream := time.Now()
+	stream, err := updatesClient.ReceiveUpdates(ctx, req)
+	assert.NoError(t, err)
+
+	_, cancel := context.WithTimeout(ctx, auth.FakeTokenValidFor+(time.Second*3))
```

Comment:
why do you create a context that you dont use at all?

Reply:
contexts are immutable

Reply:
why would it? the context that is being canceled is not used, and nothing else will be triggered that would warrant a failed test

Reply:
Now that you mention it ... weird that I tested this behaviour before and it worked. _Fixed._